test/lib/absinthe_test.exs

```diff
@@ -200,6 +200,19 @@ defmodule AbsintheTest do
     assert_result {:ok, %{data: %{"thing" => %{"name" => "Foo", "value" => 100}}}}, result
   end
 
+  it "enforces non_null fields in input passed as variable" do
+    query = """
+    mutation UpdateStuff($input: InputStuff!) {
```

Comment:
Might not, I was porting this from one of our tests, where it happened to be a mutation. I'll see and simplify to a query if that's the case.

Should I create an Issue for tracking?